File: LINKED-LIST-AND-MATRIX-EXERCISE/PYTHON/EXERCISE-2.py
#  EXERCISE 2 
# reverse() - Reverse and print out all node values of a linked list data

class Node:
    def __init__(self, data):
        self.data = data
        self.next = None

class LinkedList:
    def __init__ (self):
        self.head = None

    def reverse(self):
        prev = None
        current = self.head
        while(current is not None):
            next = current.next
            current.next = prev 
            prev = current 
            current = next
        self.head = prev

    def printList(self):
        temp = self.head
        while(temp):
            print (temp.data)
            temp = temp.next

File: LINKED-LIST-AND-MATRIX-EXERCISE/DART/EXERCISE-2.dart
// EXERCISE 2 
// reverse() - Reverse and print out all node values of a linked list data

import 'dart:io';

class Node {
  int data;
  Node next;
  Node(int data) {
    this.data = data;
    this.next = null;
  }
}

class LinkedList {
  Node head;
  LinkedList() {
    this.head = null;
  }

  void insert(int data) {
    Node newNode = new Node(data);
    if (this.head == null) {
      this.head = newNode;
    } else {
      Node last = this.head;
      while (last.next != null) {
        last = last.next;
      }
      last.next = newNode;
    }
  }

  void reverse() {
    Node prev = null;
    Node current = this.head;
    Node next = null;
    while (current != null) {
      next = current.next;
      current.next = prev;
      prev = current;
      current = next; 
    }
    this.head = prev;
  }

  void printList() {
    Node last = this.head;
    while (last != null) {
      stdout.write("${last.data} ");
      last = last.next;
    }
    print("");
  }
}